deployments/ooi/config/staging.yaml

```diff
@@ -13,3 +13,10 @@ pangeo:
       service:
         annotations: {'service.beta.kubernetes.io/azure-load-balancer-resource-group': 'ooi-pangeo'}
         loadBalancerIP: 13.92.63.24
+  dask-gateway:
+    webProxy:
+      service:
+        loadBalancerIP: 13.82.106.226
+    schedulerProxy:
+      service:
+        loadBalancerIP: 52.168.133.15
```

Comment:
Am I understanding correctly that labeling these and creating DNS entries is only necessary for bypassing JupyterHub https? Or is this necessary to simplify the user-facing dashboard addresses?

Reply:
As someone not so familiar with networking I'm always unsure what IP addresses should be public or not... so as a precaution (in case authentication and https isn't properly set up) I'm leaning towards hiding these to the extent possible. So how about putting loadBalancerIPs under secrets/staging.yaml instead of config/staging.yaml ?

Reply:
Well, as far as I know, users will need the proxy IP address to create a gateway, and having a DNS entry might provide a better user experience:
```
gateway = Gateway(address='https://staging.ooi.pangeo.io/services/dask-gateway',
                  proxy_address='tls://52.168.133.15:8786',
                  auth='jupyterhub')
```
I don't know if this can be obscured. But maybe. Also the best user experience is with the extension, so maybe there is a way to automatically populate the extension. Regarding secrets, I don't see any reason why these IP addresses would need to be secret. @jcrist do you have any thoughts/guidance on this? Thanks!

Reply:
If you're connecting from inside the kubernetes cluster (which is true if you're running on JupyterHub) then you can use the DNS entry that kubernetes created automatically for this service. Something like:

```
proxy_address="tls://scheduler-public-ooi-prod-dask-gateway:8786"
```

To keep things easy for users, you can stash these values in dask's user config (I recommend doing this). The dev instance does this here: https://github.com/pangeo-data/pangeo-cloud-federation/blob/e55e623ba9bcbd506f3a5c40e65f69a9a2689fd3/deployments/dev/image/binder/dask_config.yaml#L47-L54. See https://gateway.dask.org/configuration-user.html#default-configuration for docs on all user-side configuration options.

Reply:
@jcrist - thanks for the additional info! One issue with putting the config into image/binder/dask_config.yaml is that many of the hubs are now using multiple images in a profile_list. It would be best to set these URLs for any image on the hub, so maybe we copy dask_config.yaml under initContainers? https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/icesat2/config/common.yaml#L16

Reply:
Could also set it via environment variable - any dask configuration value can be set by a corresponding environment variable (see https://docs.dask.org/en/latest/configuration.html). For example:

```
DASK_GATEWAY__ADDRESS="https://staging.hub.pangeo.io/services/dask-gateway/"
```

These could be added via the pod spec, so they'd apply to any image.

Reply:
Just FYI, https://zero-to-jupyterhub.readthedocs.io/en/latest/customizing/user-environment.html#set-environment-variables may be helpful here. I'm trying that out on the binder deployment now (https://github.com/pangeo-data/pangeo-binder/pull/110/files)

Reply:
(obviously, valid syntax helps: https://github.com/pangeo-data/pangeo-binder/pull/111 😄)